### DISCUSSION TITLE
Deep copy with AutoMapper

### DISCUSSION THREAD

**Discussion by Kebechet** [LuckyPennySoftware/AutoMapper#4107] [NA]
I want to do a deep copy of an object. I expected that when I map object it will return me new instance but on my surprise it returned object by reference.

In my profile file I have:
`CreateMap<TrainingSetDto, TrainingSetDto>()`

and in the code I wanted to use mapping like this (`_mapper` variable is from DI):
`var shouldBeCopy= _mapper.Map<TrainingSetDto>(trainingSet);`

How to achieve my wanted result to get new instance with same properties ? E.g. on this 
blog https://jshowers.com/create-deep-copies-of-object-in-c-using-automapper/
was mentioned that static call `Mapper.Map<Person>(originalPerson);` is the way how to do so. But these static methods are not there anymore. Could you please point me to the right direction ?

Thank you.

**Comment by jbogard** [comment]
AutoMapper is not a cloning library. It is not designed to do so. Do not use AutoMapper to clone objects, pick another approach.

**Reply by Kebechet** [reply]
I agree that it is not cloning library per se... but it contains all functionality required for this usecase as well, doesnt it ?
Plus from the blog linked there it seemed possible in the past, so I am interested if it is still possible but just with different syntax.

**Reply by jbogard** [reply]
Your surprise in the behavior that AutoMapper prefers to copy references to objects of the same type should be a clue that it's not designed for cloning. It might work in some cases with extra configuration, but it's not the intended use case. A solution explicitly designed for cloning is a better option.
